src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyDescriptorGridEntry.PropertyDescriptorGridEntryAccessibleObject.cs

```diff
@@ -62,24 +163,103 @@ internal override void Expand()
+                    {
+                        if (child == dropDownButton.AccessibilityObject)
+                        {
+                            return index;
+                        }
+
+                        index++;
+                    }
+                    else if (propertyGridView.DialogButton is { Visible: true } dialogButton)
+                    {
+                        if (child == dialogButton.AccessibilityObject)
+                        {
+                            return index;
+                        }
+
+                        index++;
+                    }
+                }
+
+                if (_owningPropertyDescriptorGridEntry is { ChildCount: > 0, Expanded: true })
+                {
+                    foreach (GridEntry childEntry in _owningPropertyDescriptorGridEntry.Children)
+                    {
+                        if (child == childEntry.AccessibilityObject)
+                        {
+                            return index;
+                        }
+
+                        index++;
+                    }
+                }
+
+                return -1;
+            }
+
+            internal AccessibleObject GetNextChild(AccessibleObject child)
             {
-                switch (direction)
-                {
-                    case UiaCore.NavigateDirection.NextSibling:
-                        var propertyGridViewAccessibleObject = (PropertyGridView.PropertyGridViewAccessibleObject)Parent;
-                        var propertyGridView = propertyGridViewAccessibleObject.Owner as PropertyGridView;
-                        return propertyGridViewAccessibleObject.GetNextGridEntry(_owningPropertyDescriptorGridEntry, propertyGridView.TopLevelGridEntries, out _);
-                    case UiaCore.NavigateDirection.PreviousSibling:
-                        propertyGridViewAccessibleObject = (PropertyGridView.PropertyGridViewAccessibleObject)Parent;
-                        propertyGridView = propertyGridViewAccessibleObject.Owner as PropertyGridView;
-                        return propertyGridViewAccessibleObject.GetPreviousGridEntry(_owningPropertyDescriptorGridEntry, propertyGridView.TopLevelGridEntries, out _);
-                    case UiaCore.NavigateDirection.FirstChild:
-                        return GetFirstChild();
-                    case UiaCore.NavigateDirection.LastChild:
-                        return GetLastChild();
-                }
-
-                return base.FragmentNavigate(direction);
+                int index = GetChildIndex(child);
+
+                if (index == -1 || index == GetChildCount() - 1)
+                {
+                    return null;
+                }
+
+                return GetChild(index + 1);
+            }
+
+            internal AccessibleObject GetPreviousChild(AccessibleObject child)
+            {
+                int index = GetChildIndex(child);
+
+                if (index == -1 || index == 0)
```

Comment:
```suggestion
                if (index <= 0)
```
Perhaps this option is easier?

Reply:
Alternatively - add a Debug.Assert